nemo/collections/nlp/models/language_modeling/megatron_finetune_model.py

```diff
@@ -552,14 +577,20 @@ def write_predictions_to_file(self, outputs, output_file_path_prefix):
     def validation_step(self, dataloader_iter, batch_idx, dataloader_idx=0):
         return self.inference_step(dataloader_iter, batch_idx, 'validation', dataloader_idx)
 
-    def validation_epoch_end(self, outputs):
-        _ = self.inference_epoch_end(outputs, 'validation', self.cfg.data.validation_ds)
+    def on_validation_epoch_end(self):
+        _ = self.inference_epoch_end(self.validation_step_outputs, 'validation', self.cfg.data.validation_ds)
+        self.validation_step_outputs.clear()
```

Comment:
is it safe to run `.clear()` on `validation_step_outputs` ?
it looks like validation_step_outputs is created as a nested list in case of multiple dataloaders in init of ModelPT. Clearing it here will make it an empty list without any sublists, possibly causing errors when you try to append by specifying indexes in `validation_step`?

Reply:
@aklife97 very good catch, thanks! It needs to be cleared as `self.validation_step_outputs[dataloader_idx].clear()`. Let me fix it

Reply:
Fixed in the latest commit.